python-client/giskard/models/base/__init__.py

```diff
@@ -281,21 +290,33 @@ class WrapperModel(BaseModel, ABC):
     model_postprocessing_function: any
 
     def __init__(
-        self,
-        clf,
-        model_type: Union[SupportedModelTypes, str],
-        data_preprocessing_function=None,
-        model_postprocessing_function=None,
-        name: str = None,
-        feature_names=None,
-        classification_threshold=0.5,
-        classification_labels=None,
+            self,
+            clf,
+            model_type: Union[SupportedModelTypes, str],
+            data_preprocessing_function=None,
+            model_postprocessing_function=None,
+            name: str = None,
+            feature_names=None,
+            classification_threshold=0.5,
+            classification_labels=None,
     ) -> None:
         super().__init__(model_type, name, feature_names, classification_threshold, classification_labels)
         self.clf = clf
         self.data_preprocessing_function = data_preprocessing_function
         self.model_postprocessing_function = model_postprocessing_function
 
+        if self.data_preprocessing_function:
```

Comment:
it can be cleaner if instead of `any` we declare a type of pre/post-processing functions using `Callable[...]` in case it's provided `@validate_arguments` might do the check itself

Reply:
`validate_arguments` doesn't seem to do it, but we can later add these lines to the custom decorator `validate_args`
<img width="605" alt="Screenshot 2023-04-07 at 13 19 15" src="https://user-images.githubusercontent.com/32709181/230600438-e0c5899f-dc8f-4cd4-a981-97c7de78fdc6.png">